include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl.h

```diff
@@ -755,7 +756,7 @@ template <typename _Type>
 bool
 __group_scan_fits_in_slm(const sycl::queue& __queue, ::std::size_t __n, ::std::size_t __n_uniform)
 {
-    constexpr int __single_group_upper_limit = 16384;
+    constexpr int __single_group_upper_limit = 2048;
```

Comment:
Is that based on some empirical finding? If so, it may sense to leave a comment.

Reply:
I've refactored this a little to add a comment, and have a separate threshold for reduce-then-scan and legacy scan.